open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/retrieval/flow/VRPConsentRetrievalUtilTest.java

```diff
@@ -0,0 +1,432 @@
+    }
+
+    @Test
+    public void testGetConsentDataSetForAccounts() {
+        Mockito.doReturn(ConsentExtensionConstants.ACCOUNTS).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.VALID_INITIATION_OBJECT).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray accountConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(accountConsentData);
+    }
+
+    @Test(dataProvider = "PaymentConsentDataDataProvider", dataProviderClass = ConsentExtensionDataProvider.class)
+    public void testGetConsentDataSetForPayments(String paymentReceipt) {
+
+        Mockito.doReturn(configMap).when(openBankingConfigParse).getConfiguration();
+
+        PowerMockito.mockStatic(OpenBankingConfigParser.class);
+        PowerMockito.when(OpenBankingConfigParser.getInstance()).thenReturn(openBankingConfigParse);
+
+        Mockito.doReturn(ConsentExtensionConstants.PAYMENTS).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(paymentReceipt).when(consentResourceMock).getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.CREATED_TIME).when(consentResourceMock)
+                .getCreatedTime();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray paymentConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(paymentConsentData);
+    }
+
+    @Test
+    public void testGetConsentDataSetForCOF() {
+
+        Mockito.doReturn(ConsentExtensionConstants.FUNDSCONFIRMATIONS).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.COF_RECEIPT).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray cofConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(cofConsentData);
+    }
+
+    @Test
+    public void testGetConsentDataSetForVRPData() {
+
+        Mockito.doReturn(ConsentExtensionConstants.VRP).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.VRP_INITIATION).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+            JSONArray vrpConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+            Assert.assertNotNull(vrpConsentData);
+    }
+
+    @Test
+    public void testAppendDummyAccountID() {
```

Comment:
I guess we dont need unit test for this method as this is dummy data. WDUT?

Reply:
Fixed in [460b4c0]